web/shared/src/typeahead.ts

```diff
@@ -99,12 +99,28 @@ export function query_matches_string_in_order(
     source_str: string,
     split_char: string,
 ): boolean {
+    query = query.toLowerCase();
     source_str = source_str.toLowerCase();
-    source_str = remove_diacritics(source_str);
 
-    query = query.toLowerCase();
-    query = remove_diacritics(query);
+    // Ignore diacritics from `source_str` only if `query` contains only ASCII characters.
+    // This ensures diacritic-sensitive queries match accurately.
+    const is_query_ascii = /^[a-z]+$/.test(query);
```

Comment:
*None* of these things are the same:

- Only lowercase letters
- Only ASCII characters
- No diacritics